api/src/utils/3d-network.js

```diff
@@ -0,0 +1,73 @@
+import createGraph from 'ngraph.graph';
+import createLayout from 'ngraph.forcelayout';
+
+const SCALE = 5;
+const MAX_ITERATIONS = 1000;
+
+const populateWithLayout = ({ nodes, links }) => {
+  const g = createGraph();
+
+  for (let node of nodes) {
+    const { id, ...data } = node;
+    g.addNode(id, data);
+  }
+
+  for (let link of links) {
+    const { s, t } = link;
+    g.addLink(s, t);
+  }
+
+  const startTime = Date.now();
+  const layout = createLayout(g, { dimensions: 3 });
+
+  let iterations = MAX_ITERATIONS;
+  const elementsCount = nodes.length + links.length;
+  if (elementsCount > 50000) {
```

Comment:
It might be that `iterations` is not the only useful parameter to tweak between different network sizes, see the [Physics simulator section in the documentation](https://github.com/anvaka/ngraph.forcelayout#physics-simulator).

Reply:
I did see this section but I thought for smaller graphs, given enough iterations the shapes looked really good. So I didn’t experiment too much with it. It might be worth trying it on larger graphs as you say though.

In general for larger graphs such as cytosol, I don’t know how practical it is to interact with the 3D graphs due to the vast number of elements. This is also why we cap the iterations at 200 to optimize for speed for really large graphs. While I’m keen on experimenting more with the layout calculation I don’t know how useful or practical it would in the end for large graphs.

Reply:
I believe increasing `springLength` would make the network appear more "airy".
In terms of performance, I suspect tweaking `timeStep`, `gravity` and `dragCoefficient` might speed things up a bit.
Perhaps it would make sense to create a separate issue for this though.

Reply:
I think those are valid points but I’m still worried about it “messing up” smaller graphs because of how similar in shape they are now to the old viewer. I do think we could add another issue for this though. I suggest we could identify a list of subsystems and compartments and automate the process by trying different params and capturing the outputs.